refactor: related changes

ml019 opened the pull request:
## Intent of Change
- Documentation (new or update to existing)
- Refactor (non-breaking change which improves the structure or operation of the implementation)

## Description
Rename `Followup Actions` to the more generic `Related Changes`.

Add additional formatting advice in relation to prerequisite and dependent PRs.

## Motivation and Context
- More consistency in references to other PRs

## How Has This Been Tested?
Review of the content in VSCode preview mode.

## Related Changes
- [x] None

ml019 committed:
refactor: related changes

Rename `Followup actions` to the more generic "Related Changes"

Add additional formatting advice in relation to prerequisite and
dependent PRs.

roleyfoley commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -15,12 +15,16 @@
+
     What changes must a consumer of this repository make in order to utilise it?
-    Are there other issues or steps that need to happen once this PR is merged?
+
+    Are there other issues or steps that need to happen BEFORE this PR is merged? Start a list of these with the keyword "Prerequisite PRs:"
+
+    Are there other issues or steps that need to happen AFTER this PR is merged? Start a list of these with the keyword "Dependent PRs:"
```
Can we remove the whitepsace? Its a lot to highlight and remove on every PR if we include whitespace. Can we make these more imperitive rather than asking the same question. Just to remove the sapce and make it obvious what needs to be done

RossMurr4y replied:
Agreed - it goes against convention and readability norms but templates lose their power the longer they are. 

If there's too much to read people (read: me) won't reliably read them. 

You also don't notice them if/when they get updated.

ml019 committed:
refactor: related changes compressed and imperative

ml019 commented:
@roleyfoley @RossMurr4y ready for review

RossMurr4y commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -15,12 +15,13 @@
-    Are there other issues or steps that need to happen once this PR is merged?
-
-    Add a checklist of items or leave the default of "None"
+## Related Changes
```
How about we make use of the same "None" default used earlier, and just provide headings for Prereqs and Dependancies - I think the names fairly clear as to the intention

```suggestion
## Related Changes
<!--- Add a checklist of items or leave the default of "None"
Are the changes mandatory (breaking) or optional?
What changes must a consumer of this repository make in order to utilise it?
-->

### Prerequisite PRs:
- None

### Dependent PRs:
- None
```

With those two headings implemented, I wonder if we still even need "Related Changes" on its own. 

- Breaking changes should be covered by the use of conventional commits.
- Required changes should fall into prerequisite or dependant PRs.
- Tangentially-related PR's/issues should be covered in description or context headings using closing-syntax.

ml019 committed:
Update .github/PULL_REQUEST_TEMPLATE.md

Co-authored-by: Ross Murray <ross.murray@protonmail.com>

ml019 committed:
refactor: reorganise related changes

Add three explicit headings under related changes for the typical things
provided.

ml019 commented:
@RossMurr4y I tweaked the related changes section but I think still kept the spirit of your suggestion?

RossMurr4y commented:
Brief yet informative 👍🏻
